.github/workflows/validate-pull-request.yml

```diff
@@ -0,0 +1,30 @@
+name: Validate pull request rules
+
+on:
+  pull_request:
+    types:
+    - opened
+    - labeled
+    - unlabeled
+    - synchronize
+
+jobs:
+  validate-pull-request:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Release cannot be source branch
+      if: ${{ github.head_ref == 'release' }}
+      run: exit error
+
+    - name: Master can only merge to release
+      if: ${{ github.head_ref == 'master' && github.base_ref != 'release' }}
```

Comment:
Should we have the converse as well? If we're merging to release it has to come from master

Reply:
Great catch. I thought this rule would take care of both cases, but thinking about it more realized that isn't the case.